### PR TITLE
feat: Add config to reset view to top on navigation

### LINKED ISSUE #10143: Auto scroll to middle or bottom of the page when navigating to new page.
**Describe the bug**
Auto scroll to middle or bottom of the page when navigating to new page.
In many situations where I navigate to a new page, the pages is displayed scrolled down.  It is disorienting for the user and moreover we can't see success/error messages after some actions.

**Tell us the version of Spartacus**

test env https://spartacusstore.cg79x9wuu9-eccommerc1-p5-public.model-t.myhybris.cloud/electronics-spa/en/USD/ (see #10030)

**Steps to reproduce:**

I reproduced on:
* iPhone 6s on iOS 14
* Chrome in iPhone 7+ view in dev tools on Mac.

I've seen this happen on pure desktop in the past too, but I can reproduce it consistently on mobile viewport.
See videos in attachment. (captured on chrome/mac)

Expected behavior
New pages appear at the top of the page as intended.

**Error videos:**

[Bug-videos.zip](https://github.com/SAP/spartacus/files/5619091/Bug-videos.zip)

### PR DESCRIPTION
**Pull request by Zeyber**
Closes: #10143

### CODE REVIEW

**Comment by Zeyber** [comment]
To test:

Set view and focus configs. Switch between BREAKPOINT values and boolean true/false to test features work as intended.

Example:
`app.module.ts`
```
provideConfig({
      keyboardFocus: {
        enableResetViewOnNavigate: [BREAKPOINT.xs, BREAKPOINT.sm],
        enableResetFocusOnNavigate: true,
      },
    }),
```

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/config/keyboard-focus.config.ts [hunk 1]
```diff
@@ -11,5 +12,10 @@ export abstract class KeyboardFocusConfig {
      *  Resets the focus back to the root `<cx-storefront>` element when a navigation is started.
```
Comment (Matejk00) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/config/keyboard-focus.config.ts]: Cosmetic issue, I think additional space at the beginning of comment is unnecessary.

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/config/keyboard-focus.config.ts [hunk 2]
```diff
@@ -11,5 +12,10 @@ export abstract class KeyboardFocusConfig {
...
      */
     enableResetFocusOnNavigate?: boolean;
+
+    /**
+     *  Resets the view back to the root `<cx-storefront>` element when a navigation is started. This can also be set to a BREAKPOINT value to only reset the view up to the provided value.
```
Comment (Matejk00) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/config/keyboard-focus.config.ts]: Same here.

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.spec.ts [hunk 3]
```diff
@@ -91,5 +119,91 @@ describe('OnNavigateFocusService', () => {
...
+    });
+
+    it('should scrollIntoView cx-storefront element given breakpoint value', async () => {
+      const breakpointService = TestBed.inject(BreakpointService);
+      spyOn(breakpointService, 'isDown').and.returnValue(of(true));
+
+      service.setResetViewOnNavigate(BREAKPOINT.sm);
+      await zone.run(() => router.navigateByUrl('/'));
+
+      expect(spyDoc).toHaveBeenCalledWith('cx-storefront');
```
Comment (Matejk00) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.spec.ts]: We are re-using this string (cx-storefront) in few places in this file, maybe can we move it to some const variable?

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.ts [hunk 4]
```diff
@@ -13,35 +15,90 @@ import { KeyboardFocusConfig } from '../config';
...
+          } else if (typeof enable === 'boolean') {
+            this.getStorefrontElement().focus();
+          }
+        });
+    }
+  }
+
+  /**
+   * Resets view back to root element `<cx-storefront>` in the DOM tree when a navigation is started.
+   * @param enable Enable or disable this feature. Set this to a BREAKPOINT value string to enable only up to the specified screen width.
```
Comment (Matejk00) [inline projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.ts]: I'm not sure how big limitation it is, but in this approach if customer would like to setup this behaviour for tablet and mobile screen sizes only (lg, md, sm) he is not able to do that.

Worth to consider might be using array of breakpoints as next alternative?

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.ts (hunk 4 above)
Comment (Zeyber) [reply projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.ts]: In this instance, the user would use the breakpoint `lg` and this works for all breakpoints from `lg` downwards (ie. `lg` `md` `sm` `xs`).

File: projects/storefrontlib/src/layout/a11y/keyboard-focus/config/keyboard-focus.config.ts (hunk 1 above)
Comment (Zeyber) [reply projects/storefrontlib/src/layout/a11y/keyboard-focus/config/keyboard-focus.config.ts]: This type of comment is needed to appear in the linter when highlighting a function.

**Review by Matejk00 — DISMISSED** [review]

**Review by Matejk00 — APPROVED** [review]

**Pull request merged**